doc/source/train/user-guides/persistent-storage.rst

```diff
@@ -1,62 +1,387 @@
+        )
+    )
+
+In this example, all files are saved to ``/mnt/cluster_storage/experiment_name`` for further processing.
+
+
+Local storage
+-------------
+
+Using local storage for a single-node cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you're just running an experiment on a single node (e.g., on a laptop), Ray Train will use the
+local filesystem as the storage location for checkpoints and other artifacts.
+Results are saved to ``~/ray_results`` in a sub-directory with a unique auto-generated name by default,
+unless you customize this with ``storage_path`` and ``name`` in :class:`~ray.train.RunConfig`.
+
+
+.. code-block:: python
+
+    from ray import train
+    from ray.train.torch import TorchTrainer
+
+    trainer = TorchTrainer(
+        ...,
+        run_config=train.RunConfig(
+            storage_path="/tmp/custom/storage/path",
+            name="experiment_name",
+        )
+    )
+
+
+In this example, all experiment results can found locally at ``/tmp/custom/storage/path/experiment_name`` for further processing.
+
+
+Using local storage for a multi-node cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. warning::
+
+    When running on multiple nodes, using the local filesystem of the head node as the persistent storage location is no longer supported.
+
+    If you save checkpoints with :meth:`ray.train.report(..., checkpoint=...) <ray.train.report>`
+    and run on a multi-node cluster, Ray Train will raise an error if NFS or cloud storage is not setup.
+    This is because Ray Train expects all workers to be able to write the checkpoint to
+    the same persistent storage location.
+
+    If your training loop does not save checkpoints, the reported metrics will still
+    be aggregated to the local storage path on the head node.
+
+    See `this issue <https://github.com/ray-project/ray/issues/37177>`_ for more information.
+
+
+.. _custom-storage-filesystem:
+
+Custom storage
+--------------
+
+If the cases above don't suit your needs, Ray Train can support custom filesystems and perform custom logic.
+Ray Train standardizes on the ``pyarrow.fs.FileSystem`` interface to interact with storage
+(`see the API reference here <https://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html>`_).
+
+By default, passing ``storage_path=s3://bucket-name/sub-path/`` will use pyarrow's
+`default S3 filesystem implementation <https://arrow.apache.org/docs/python/generated/pyarrow.fs.S3FileSystem.html>`_
+to upload files. (`See the other default implementations. <https://arrow.apache.org/docs/python/api/filesystems.html#filesystem-implementations>`_)
+
+Implement custom storage upload and download logic by providing an implementation of
+``pyarrow.fs.FileSystem`` to :class:`RunConfig(storage_filesystem) <ray.train.RunConfig>`.
+
+.. warning::
+
+    When providing a custom filesystem, the associated ``storage_path`` is expected
+    to be a qualified filesystem path *without the protocol prefix*.
```

Comment:
Did we check this and raise an error in the code?

Reply:
yep, pyarrow will raise an error directly